### ISSUE TITLE
In API, ScheduledStatus params[sensitive] returning string instead of boolean

### ISSUE THREAD

**Issue by legoktm** [mastodon/mastodon#24645]
### Steps to reproduce the problem

1. Schedule a status in the future
2. Make an API request like `curl -H 'Authorization: Bearer {token}' https://mastodon.example/api/v1/scheduled_statuses`

### Expected behaviour

As per https://docs.joinmastodon.org/entities/ScheduledStatus/#params-sensitive the `sensitive` parameter should be a boolean or null

### Actual behaviour

`"sensitive" : "0",`

### Detailed description

The full response I got:
```json
[
   {
      "id" : "13",
      "media_attachments" : [],
      "params" : {
         "application_id" : 311,
         "idempotency" : null,
         "in_reply_to_id" : null,
         "language" : null,
         "media_ids" : null,
         "poll" : null,
         "scheduled_at" : null,
         "sensitive" : "0",
         "spoiler_text" : null,
         "text" : "what what",
         "visibility" : "direct",
         "with_rate_limit" : false
      },
      "scheduled_at" : "2023-04-25T16:42:00.000Z"
   }
]
```

### Specifications

Mastodon 4.1.2

**Comment by technicat** [comment]
I just ran into a variation of this where sensitive is string "false" instead of bool false. This is on universeodon.com which I believe is running 4.2.0

**Comment by ManVTish** [comment]
I agree with @trwnh , it's directly pulling from database, which could mean that it was stored incorrectly in the first place. @trwnh would it make sense to add some sort of validation.

**Comment by trwnh** [comment]
the database format is fine, it's the conversion to API format that's missing

**Comment by gustavo-guerreros1** [comment]
I hope this message finds you well. I am a student at the Universidad Nacional Mayor de San Marcos in Peru. I am writing to kindly ask for permission to fix issue #24645 in the Mastodon repository related to the API and "ScheduledStatus" parameters returning a string instead of a boolean value for "sensitive".

As students at the Universidad Nacional Mayor de San Marcos, we are very interested in contributing to open source projects and improving our experience in software development. We have been closely following the development of Mastodon and would like to help resolve this particular issue. We are comfortable working with the code and have previous experience solving similar problems on other projects.

We are committed to following Mastodon's development guidelines and working collaboratively with their team to ensure that any changes made are compatible and benefit the community at large. We are excited to have the opportunity to contribute to this project and help improve the functionality of the API.

We appreciate your consideration and look forward to getting your permission to begin work on this issue. Please let us know if you need any further information or if there are any additional procedures we need to follow.

Thank you very much for your attention and we look forward to contributing positively to the development of Mastodon!

**Comment by trwnh** [comment]
@gustavo-guerreros1 you do not need permission; you can just do it

**Comment by connyduck** [comment]
I'm encountering this as well, a fix would be very much appreciated.
